mongodb/vibe/db/mongo/collection.d

```diff
@@ -493,6 +497,46 @@ struct MongoCollection {
 		enforce(reply["ok"].get!double == 1, "dropIndex command failed: "~reply["errmsg"].opt!string);
 	}
 
+	/**
+		Creates indexes on the collection.
+	*/
+	void createIndex(T)(T query) 
```

Comment:
ensureIndex already exists, which is basically this command but using the old API. Is there a reason for this new API? I think there are new index types which we need to support but I don't really like the current API you implemented for this (accepting literally anything as argument, which is called unintuitively "query")

Reply:
https://docs.mongodb.com/manual/reference/method/db.collection.ensureIndex/#db.collection.ensureIndex ensureIndex is deprecated

Reply:
the current ensureIndex implementation actually directly writes into the indices document in mongodb, so it's really just what it's called in vibe.d currently. I don't really like this approach but maybe extending and improving ensureIndex would be a better idea than creating this new API.

Also both methods should add tests!

Reply:
imo it should be high priority to reflect mongodb's standard API layout within vibe-d's mongodb implementation so that it's easy for people to understand it.

Reply:
@WebFreak001 I can no longer use the ensureIndex method with my mongodb server (version 4.2.1) as the indexes are no longer positioned at the place ensureIndex is trying to write to. Please make the standard createIndex available as there is no reason to re-invent the wheel here.